### DISCUSSION TITLE
How to remove a tag that without some attribute value

### DISCUSSION THREAD

**Discussion by SettingDust** [apostrophecms/sanitize-html#589] [Q&A] [answered]
I want to remove tag that don't has specific attribute with value
Have tried:
   - Add to `allowedTags` and `allowedAttributes`. (Will keep all tag and just remove the attributes not exists)
   - Don't add to `allowedTags` but `allowedAttributes`. (Won't add)

**Comment by BoDonkey** [comment, marked as answer]
You should be able to do this with `exclusiveFilter`. 
```
sanitizeHtml(
  '<p desired-att="desiredValue">This is the test HTML</p>',
  {
    exclusiveFilter: function(frame) {
      return !frame.attribs.hasOwnProperty('desired-att') || frame.attribs['desired-att'] !== desiredValue;
    }
  }
);
```
Basically, just write a function that returns `true` if the section of HTML should be removed. In the code above, I'm returning true if the  `attribs` object doesn't have a key that is equal to the attribute that I want present or if that attribute doesn't have a specific value.
Hope this helps!

**Reply by SettingDust** [reply]
Why is the arg named `frame`. I mistakenly thought it's relating to frame

**Reply by BoDonkey** [reply]
Naming things can be hard and if you look at the file, it has been through the hands of a lot of people. Sometimes consistency gets a little lost. Maybe there was an example before it using an `iframe`. You can open a PR to correct it if you think it would be helpful. Thanks!

**Comment by boutell** [comment]
It's called "frame" because there is a stack of tags happening which can be
thought of as nested "frames" internally.

On Fri, Dec 9, 2022 at 5:23 AM Robert Means ***@***.***>
wrote:

> Naming things can be hard and if you look at the file, it has been through
> the hands of a lot of people. Sometimes consistency gets a little lost.
> Maybe there was an example before it using an iframe. You can open a PR
> to correct it if you think it would be helpful. Thanks!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/apostrophecms/sanitize-html/discussions/589#discussioncomment-4350873>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAH27KWYCEOUSA3TZ3SEU3WMMCDLANCNFSM6AAAAAASXW4PGI>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: <apostrophecms/sanitize-html/repo-discussions/589/comments/4350873@
> github.com>
>


-- 

THOMAS BOUTELL | CHIEF TECHNOLOGY OFFICER
APOSTROPHECMS | apostrophecms.com | he/him/his

**Comment by boutell** [comment]
(Which is not to say Bob's assessment isn't 100% accurate too, from another
perspective)

On Tue, Dec 13, 2022 at 10:44 AM Tom Boutell ***@***.***> wrote:

> It's called "frame" because there is a stack of tags happening which can
> be thought of as nested "frames" internally.
>
> On Fri, Dec 9, 2022 at 5:23 AM Robert Means ***@***.***>
> wrote:
>
>> Naming things can be hard and if you look at the file, it has been
>> through the hands of a lot of people. Sometimes consistency gets a little
>> lost. Maybe there was an example before it using an iframe. You can open
>> a PR to correct it if you think it would be helpful. Thanks!
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/apostrophecms/sanitize-html/discussions/589#discussioncomment-4350873>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AAAH27KWYCEOUSA3TZ3SEU3WMMCDLANCNFSM6AAAAAASXW4PGI>
>> .
>> You are receiving this because you are subscribed to this thread.Message
>> ID: <apostrophecms/sanitize-html/repo-discussions/589/comments/4350873@
>> github.com>
>>
>
>
> --
>
> THOMAS BOUTELL | CHIEF TECHNOLOGY OFFICER
> APOSTROPHECMS | apostrophecms.com | he/him/his
>


-- 

THOMAS BOUTELL | CHIEF TECHNOLOGY OFFICER
APOSTROPHECMS | apostrophecms.com | he/him/his
